pgcli/packages/iospecial.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import print_function
+
+
+import logging
+import click
+
+_logger = logging.getLogger(__name__)
+
+
+def editor_command(command):
+    """
+    Is this an external editor command?
+    :param command: string
+    """
+    return command.strip().startswith('\e')
+
+
+def open_external_editor(filename=None):
+    """
+    Open external editor, wait for the user to type in his query,
+    return the query.
+    :return: list with one tuple, query as first element.
+    """
+
+    message = None
+    filename = filename.strip().split(' ', 1)[0] if filename else None
+
+    MARKER = '# Type your query above this line.\n'
+    query = click.edit('\n\n' + MARKER, filename=filename)
```

Comment:
Feed the partial query to `click.edit()` so it is auto-populated into the editor buffer.